### PR TITLE
Identity: extend k8s user.Info

### PR DESCRIPTION
**Pull request by ryantxu**
This PR refactors the identity.Requester so it extends k8s user.Info

The biggest implication is that `GetUID()` must return a string -- so we moved the typed version to `GetTypedUID()`

### CODE REVIEW

**Comment by charandas** [comment]
LGTM - however, I do have a question on the sibling enterprise PR. In the meanwhile, I will run with it locally to ensure anonymous access still works as intended for metrics scraping and profiles.

**Review by charandas — APPROVED** [review]

File: pkg/services/apiserver/auth/authenticator/signedinuser.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ func signedInUserAuthenticator(req *http.Request) (*authenticator.Response, bool
 
 	userInfo := &k8suser.DefaultInfo{
```
Comment (charandas) [inline pkg/services/apiserver/auth/authenticator/signedinuser.go:24]: Why don't we return simply return the `requester` here?

**Comment by ryantxu** [comment]
@charandas -- I removed https://github.com/grafana/grafana-enterprise/pull/6937, we can approach that after this PR is resolved

File: pkg/storage/unified/apistore/prepare.go [hunk 2]
```diff
@@ -81,7 +81,7 @@ func (s *Storage) prepareObjectForUpdate(ctx context.Context, updateObject runti
 		return nil, err
 	}
 	obj.SetOriginInfo(origin)
-	obj.SetUpdatedBy(user.GetUID().String())
+	obj.SetUpdatedBy(user.GetUID())
```
Comment (ryantxu) [inline pkg/storage/unified/apistore/prepare.go:84]: @Jguer @gamab -- this is the most visible change.  k8s requires "GetUID()" to be a string, and we currently return a typed flavor

**Comment by Jguer** [comment]
@charandas did you confirm anon access still works? 

I think that is usually the most break prone component

**Comment by charandas** [comment]
> @charandas did you confirm anon access still works?
> 
> I think that is usually the most break prone component

It works in standalone / apiserver mode we have been using requester/contexthandler from Grafana OSS.

As for verifying whether it works in OSS itself, I tested with Dashboard Snapshots and it works in the UI. There is a 401 that I also see on `main` related to not being able to fetch `http://localhost:3000/api/prometheus/grafana/api/v1/rules?dashboard_uid=<some_guid>` which I disregarded for the comparison.

Public dashboards are another place I confirmed it also works with anonymous access.

**Pull request merged**